services/api/lib/postgre.js

```diff
@@ -295,6 +295,334 @@ module.exports = function (pg) {
         });
       });
 
+      // matches string like '$0.id' where '$0' is the action result at the 0th index and the value on that object keyed with 'id'
```

Comment:
regex + the replace function, really, so we can do something like var replace = require('./replacer') (optionally with some args if it needs to bind closure values)